### PR TITLE
Initial implementaion

### PR DESCRIPTION
**Pull request by oliverfoster**

### CODE REVIEW

**Comment by brian-learningpool** [comment]
There are still some `console.log()` calls in there. 

Also, the properties.schema file should be updated for compatibility with the authoring tool.

File: js/adapt-contrib-bookmarking.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+                ],
+                _showIcon: true
+            }
+
+            Adapt.trigger('notify:prompt', promptObject);
+        },
+
+        navigateToPrevious: function() {
+            var courseBookmarkModel = Adapt.course.get('_bookmarking');
+            var targetModel = Adapt.findById(courseBookmarkModel._locationID);
```
Comment (moloko) [inline js/adapt-contrib-bookmarking.js]: this line (89) looks as if it is now redundant...

**Comment by dancgray** [comment]
+1, seems good but bookmarking will rely on being able to store question selection at least in order to enable question re-attempts.

**Comment by brian-learningpool** [comment]
+1

**Comment by himanshu1618** [comment]
+1

File: js/adapt-contrib-bookmarking.js [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+                $(window).on("scroll", this._onScroll);
+                this.listenTo(Adapt, this.bookmarkLevel + "View:postRender", this.addInViewListeners);
+                this.listenToOnce(Adapt, "remove", this.removeInViewListeners);
+            }
+        },
+
+        addInViewListeners: function (view) {
+            var element = view.$el;
+            element.data('locationID', view.model.get('_id'));
+            console.log("View check on", view.model.get('_id'));
```
Comment (moloko) [inline js/adapt-contrib-bookmarking.js:131]: need to remove this

**Pull request merged**